datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -203,6 +203,24 @@ impl LogicalPlanBuilder {
         Self::scan_with_filters(table_name, table_source, projection, vec![])
     }
 
+    /// Convert a logical plan into a builder with a [DmlStatement]
+    pub fn insert_into(
+        input: LogicalPlan,
+        table_name: impl Into<String>,
```

Comment:
I think this will (logically) conflict with changes @Jefffrey  is making in https://github.com/apache/arrow-datafusion/pull/5343

Reply:
I don't have complete knowledge about the PR at the moment, but I'll look into it. Nevertheless, it's not a critical inclusion, so it can be readily eliminated if necessary.

Reply:
I think it would just involve changing from:

```rust
table_name: impl Into<String>
```

to:

```rust
table_name: impl Into<OwnedTableReference>
```

and use that directly, if #5343 gets merged first. Since there will be a `From<String>` implementation for `OwnedTableReference` anyway

Reply:
Addressed 👍🏻